pkgs/unified_analytics/lib/src/log_handler.dart

```diff
@@ -160,17 +161,23 @@ class LogHandler {
   final FileSystem fs;
   final Directory homeDirectory;
   final File logFile;
+  final Analytics _analyticsInstance;
+
+  /// Ensure we are only sending once per invocation for this class.
```

Comment:
Why only send once?

Reply:
In the log handlers case, we are checking if each log record is malformed. So in the worst case scenario, we could have 2,500 malformed log records. This check would it make it so that we don't receive tons of errors logs from just one client

Code from log handler below where send these events

```dart
    final records = logFile
        .readAsLinesSync()
        .map((String e) {
          try {
            return LogItem.fromRecord(jsonDecode(e) as Map<String, Object?>);
          } on FormatException catch (err) {
            if (!_errorEventSent) {
              _sendFunction(Event.analyticsException(
                workflow: 'LogFileStats.logFileStats',
                error: err.runtimeType.toString(),
                description: 'message: ${err.message}\nsource: ${err.source}',
              ));
              _errorEventSent = true;
            }
            return null;
            // ignore: avoid_catching_errors
          } on TypeError catch (err) {
            if (!_errorEventSent) {
              _sendFunction(Event.analyticsException(
                workflow: 'LogFileStats.logFileStats',
                error: err.runtimeType.toString(),
              ));
              _errorEventSent = true;
            }
            return null;
          }
        })
        .whereType<LogItem>()
        .toList();
```

Reply:
Oh that is a good point, as it stands right now, it would be per dart process.  In that case, i can rework the logic for `_errorEventSent` so that it gets reset every `x` minutes. The main purpose of this was so that one client wasn't reporting hundreds of the same error event

Reply:
Rather than using timestamps, can we rather group together operations that we expect could error multiple times, and track a single local bool errorAlreadySent for those operations?

Reply:
I think it might be even better to create an abstraction for handling errors, something like `ErrorHandler` that can be passed to each class created by `Analytics`.

This class can keep track of what events we have sent for errors and block any events we have already sent. That should centralize our error handling to one class